### PR TITLE
Workflowv2 workflows get

### PR DESCRIPTION
**Pull request by garbage-collector**
For #1199 

https://github.com/openstack/mistral/blob/master/mistral/api/controllers/v2/workflow.py#L80

### CODE REVIEW

**Comment by garbage-collector** [comment]
Ready to review!

**Comment by jtopjian** [comment]
This one also looks good, but I'll take another look once everything has been cleaned up.

**Comment by garbage-collector** [comment]
I cleaned up and rebased everything in one commit

File: acceptance/openstack/workflow/v2/workflows_test.go [hunk 1]
```diff
@@ -8,12 +8,27 @@ import (
...
 	th.AssertNoErr(t, err)
 
 	workflow, err := CreateWorkflow(t, client)
 	th.AssertNoErr(t, err)
+	defer DeleteWorkflow(t, client, workflow)
 
 	tools.PrintResource(t, workflow)
 }
+
+func TestWorkflowsGet(t *testing.T) {
```
Comment (jtopjian) [inline acceptance/openstack/workflow/v2/workflows_test.go]: You can probably just add the Get call within the `TestWorkflowsCreateDelete` test. It'll save some duplicated testing.

File: openstack/workflow/v2/workflows/results.go [hunk 2]
```diff
@@ -21,6 +21,27 @@ func (r CreateResult) Extract() ([]Workflow, error) {
...
+// GetResult is the response of Get operations. Call its Extract method to interpret it as a Workflow.
+type GetResult struct {
+	commonResult
+}
+
+// Extract helps to get a Workflow struct from a Get function.
+func (r commonResult) Extract() (*Workflow, error) {
+	c := Workflow{}
+	err := r.ExtractInto(&c)
+	return &c, err
```
Comment (jtopjian) [inline openstack/workflow/v2/workflows/results.go]: The normal pattern for this is:

```go
func (r GetResult) Extract() (*Workflow, error) {
  var s Workflow
  err := r.ExtractInto(&s)                        
  return &s, err                                  
}                                                 
```

**Comment by garbage-collector** [comment]
Conflicts fixed and reviews also :)

File: openstack/workflow/v2/workflows/requests.go [hunk 3]
```diff
@@ -60,3 +60,10 @@ func Delete(client *gophercloud.ServiceClient, id string) (r DeleteResult) {
 	_, r.Err = client.Delete(deleteURL(client, id), nil)
 	return
 }
+
+// Get retrieves details of a single execution.
+// Use ExtractWorkflow to convert its result into an Workflow.
```
Comment (jtopjian) [inline openstack/workflow/v2/workflows/requests.go]:
```go
// Use Extract to convert its result into a Workflow
```

**Pull request merged**